graylog2-web-interface/src/components/common/KeyCapture.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useEffect } from 'react';
+import Mousetrap from 'mousetrap';
+
+type Props = {
+  children?: React.ReactElement,
+  keys: { [key: string ]: () => void },
+}
+
+const KeyCapture = ({ children, keys } : Props) => {
+  useEffect(() => {
+    Object.entries(keys).forEach(([key, onKeyPress]) => {
+      Mousetrap.bind(key, onKeyPress);
+    });
+
+    return () => {
+      Mousetrap.reset();
```

Comment:
> The reset method will remove anything you have bound to mousetrap.

This concerns me. What happens when we have multiple instances of `KeyCapture` used in any given render? Will this reset all different instances or do it know to only reset the existing ones within this single instance? 

Maybe we could extract the similar `bind` loop and allow for unbind of the same keys instead of reseting all keys?

Reply:
Good point, we still have to reset all bindings for a key (https://github.com/ccampbell/mousetrap/issues/280) But it is still definitely better than resetting all bindings for all keys.